src/script/view_model/bindings/MessageListBindings.js

```diff
@@ -76,6 +79,66 @@ ko.bindingHandlers.showAllTimestamps = {
   },
 };
 
+ko.bindingHandlers.infinite_scroll = {
+  init(scrollingElement, params) {
+    const {onHitTop, onHitBottom} = params();
+
+    let scrollListenerEnabled = false;
+    const disableScrollListener = debounce(() => (scrollListenerEnabled = false), 500);
+    const temporarlyEnableScrollListener = () => {
+      scrollListenerEnabled = true;
+      disableScrollListener();
+    };
+
+    const onScroll = event => {
+      if (!scrollListenerEnabled) {
+        return;
+      }
+      const element = event.target;
+
+      // On some HDPI screen scrollTop returns a floating point number instead of an integer
+      // https://github.com/jquery/api.jquery.com/issues/608
+      const scrollPosition = Math.ceil(element.scrollTop);
+      const scrollEnd = element.offsetHeight + scrollPosition;
+      const hitTop = scrollPosition <= 0;
+      const hitBottom = scrollEnd >= element.scrollHeight;
+
+      if (hitTop) {
+        onHitTop();
+      } else if (hitBottom) {
+        onHitBottom();
+      }
+    };
+
+    const onMouseWheel = event => {
+      /**
+       * To avoid reacting to programmatically triggered scroll, we just want to enable the scroll listener when there is a mouse event before it.
+       * The strategy here is to enable the scroll listener when there is a mouse event and automatically disable the listener after 500ms of inactivity of the mouse.
+       */
```

Comment:
I think it might be problematic to only react to the `mousewheel` event. There are other user interactions that will cause scrolling, selecting chunks of text and dragging the scrollbar for example.